pkg/shell/index.js

```diff
@@ -336,14 +362,20 @@ define([
         }
 
         if (address) {
+            machine = machines.lookup(address);
 
             /* If the machine is not available, then redirect to dashboard */
-            machine = machines.lookup(address);
-            if (!machine) {
-                if (listing)
+            if (!machine || machine.problem) {
+                if (!machine && !machines.read_complete) {
+                    show_connecting();
+                } else if (machine && reconnect && machine.problem) {
```

Comment:
Shouldn't we track this off of the machines object and not expect the value to be passed into navigate()?